src/descriptors.rs

```diff
@@ -162,42 +171,62 @@ fn csv_check(csv_value: u32) -> Result<(), DescCreationError> {
         .map_err(|_| DescCreationError::InsaneTimelock(csv_value))
 }
 
-fn is_unhardened_deriv(key: &descriptor::DescriptorPublicKey) -> bool {
+// We require the descriptor key to:
+//  - Be deriveable (to contain a wildcard)
+//  - Be multipath (to contain a step in the derivation path with multiple indexes)
+//  - The multipath step to only contain two indexes, 0 and 1.
+fn is_valid_desc_key(key: &descriptor::DescriptorPublicKey) -> bool {
     match *key {
-        descriptor::DescriptorPublicKey::Single(..) => false,
-        descriptor::DescriptorPublicKey::XPub(ref xpub) => {
+        descriptor::DescriptorPublicKey::Single(..) | descriptor::DescriptorPublicKey::XPub(..) => {
+            false
+        }
+        descriptor::DescriptorPublicKey::MultiXPub(ref xpub) => {
+            // Rust-miniscript enforces BIP389 which states that all paths must have the same len.
+            let len = xpub.derivation_paths.get(0).expect("Cannot be empty").len();
             xpub.wildcard == descriptor::Wildcard::Unhardened
+                && xpub.derivation_paths.len() == 2
+                && xpub.derivation_paths[0][len - 1] == 0.into()
+                && xpub.derivation_paths[1][len - 1] == 1.into()
         }
     }
 }
 
+/// An [InheritanceDescriptor] that contains multipath keys for (and only for) the receive keychain
+/// and the change keychain.
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct MultipathDescriptor {
+    multi_desc: descriptor::Descriptor<descriptor::DescriptorPublicKey>,
+    receive_desc: InheritanceDescriptor,
+    change_desc: InheritanceDescriptor,
+}
+
 /// A Miniscript descriptor with a main, unencombered, branch (the main owner of the coins)
-/// and a timelocked branch (the heir).
+/// and a timelocked branch (the heir). All keys in this descriptor are singlepath.
 #[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
 pub struct InheritanceDescriptor(descriptor::Descriptor<descriptor::DescriptorPublicKey>);
 
 /// Derived (containing only raw Bitcoin public keys) version of the inheritance descriptor.
 #[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
 pub struct DerivedInheritanceDescriptor(descriptor::Descriptor<DerivedPublicKey>);
 
-impl fmt::Display for InheritanceDescriptor {
+impl fmt::Display for MultipathDescriptor {
```

Comment:
Can I use this method in the installer to fill the config descriptor field ?

Reply:
I guess yes: https://github.com/revault/minisafe/pull/74/files#diff-cba64c21ab992eaad29fce147a08f4560a4769bc14682b8a96081a5fd02dbecdR93